apps/web/src/state/farmsV4/state/poolApr/fetcher.ts

```diff
@@ -147,9 +148,18 @@ const calcV3PoolApr = ({
   }
 }
 
-export const getUniversalBCakeWrapperForPool = (pool: PoolInfo) => {
+export const getUniversalBCakeWrapperForPool = (pool: { lpAddress: Address; chainId: number; protocol?: Protocol }) => {
+  let { lpAddress } = pool
+  if (pool.protocol === 'stable') {
+    const stablePair = LegacyRouter.stableSwapPairsByChainId[pool.chainId].find((pair) => {
+      return isAddressEqual(pair.stableSwapAddress, pool.lpAddress)
+    })
+    if (stablePair) {
+      lpAddress = stablePair.liquidityToken.address
```

Comment:
there are something easy to cause problem if someone don't have context (I means pool.lpAddress of stablePool is not the real lpAddress, it's the stableSwapAddress), i think we can fix the PoolInfo field

Reply:
yes, it should be more clearly, I have stuck with this part many times